### ISSUE TITLE
Pytests don't run locally

### ISSUE THREAD

**Issue by ivalaginja** [lorenaacuna/GASTLI#7]
For https://github.com/openjournals/joss-reviews/issues/7288.

@lorenaacuna unfortunately my issue with the pytests persists.

I have a conda environment that contains both `gastli` and its dependencies, as well as `pytest`. However, when I cd into the test directory of the GASTLI repo and run `pytest`, nothing happens.

![Screenshot 2024-12-02 at 17 44 59](https://github.com/user-attachments/assets/7dfc2b51-6f4b-4d8a-af9f-e20733eb33aa)

**Comment by lorenaacuna** [comment]
Hi @ivalaginja, the only difference I notice between your terminal output and mine is the line 
`plugins: anyio-4.6.2.post1`. I've made a quick Google search and anyIO is a new package that provides a plugin for testing with pytest. This is the first time I hear of anyIO, so none of the configuration files in GASTLI uses this package. Maybe it is your pytest/Python configuration that defaults to using the plugin. In any case, you can try fixing it by installing anyIO with `pip install anyio` in the same environment where you have GASTLI and pytest installed: https://anyio.readthedocs.io/en/stable/basics.html.

After installing anyIO, try running `pytest` from the test directory again. Let me know if that works.

**Comment by ivalaginja** [comment]
I don't think I would have made any global pytest settings like this on my machine. In any case, if I try to resintall wih `pip install anyio`, it just tells me that this requirement is already satisfied and I get the same result when trying to run the pytests :/

**Comment by lorenaacuna** [comment]
Can you try uninstalling it (`pip uninstall anyio` or `python -m pip uninstall anyio`) and run the test again?

**Comment by ivalaginja** [comment]
Even after uninstalling I still get the same result...
```bash
platform darwin -- Python 3.10.15, pytest-8.3.4, pluggy-1.5.0
rootdir: /Users/ilaginja/repos/GASTLI
configfile: pyproject.toml
collecting ... (GASTLI_env) macbookiv:test ilaginja$ 
```

**Comment by ivalaginja** [comment]
I figured it out, the reason the pytests wouldn't run is because I wasn't aware that I needed to edit each test file manually to include my own path to files in them. Once that was done, the tests ran fine.

However, this is really not a good practice, so we will have to find a solution for this. A suite of pytests should run out of the box, with maybe a centralized configuration needing prior updates, but definitely not each individual test script. The latter is very intransparent to users and massively prone to "stupid" mistakes like just forgetting a path somewhere. It also doesn't help that the variable that holds the path to the files is currently not called the same everywhere, so it's impossible to search the project for all locations where the path would need updating. I already noticed this in the quickstart tutorials in the documentation.

Moreover, it is currently necessary to have a trailing slash at the end of the path to files, otherwise it is not joined correctly in the core of the gastli package. This should be fixed programmatically, otherwise it is likely that there will be both confusion and also OS-specific errors appearing.

To fix this, I would like to motivate you to do the following two things:
- Change the internal handling of the input file path. It should accept both a path with and without a trailing slash. This is easily achievable with the `os` package, using functions like `os.join()`, which adds a required slash dynamically if it is not present in the input path.
- Find a way to set up the local path to the input files only once on a user's local machine. There's a lot of fancy solutions out there with configuration files of different formats, but I think in your case, you could simply use an environment variable. And then you include instructions for the setup of that environment variable in your installation instructions, in the part where you tell people to download the data. Then, you replace each place in your code where the path should be put in with simply reading that environment variable, especially the tests and the quickstart tutorials.

**Comment by lorenaacuna** [comment]
@ivalaginja thanks so much for catching this and the suggestions! It totally makes sense to handle it this way for the pytests to work. 

I modified the source code, examples and tests to handle the input path with an environment variable. I updated the documentation to include a subsection about how to define the environment variable in the installation page. 

To check the new feature and that the pytests work without having to modify the scripts: 
1. Install GASTLI's latest version, 0.9.3 (`pip install gastli`, or `pip install gastli=0.9.3`) -  this is the one that handles the path with an environment variable.
2. Follow the installation page in the docs to define the env. variable `GASTLI_input_data_path`
3. Run the pytests as usual 

Please let us know if the pytests work so I can close this issue (#7) and issue #6.

**Comment by ivalaginja** [comment]
Tests pass now without needing to adjust paths! This issue can be closed, however, please hold off on closing #6 for now.
